feature-libs/pickup-in-store/components/container/pdp-pickup-options-container/pdp-pickup-options-container.component.ts

```diff
@@ -173,16 +179,36 @@ export class PdpPickupOptionsContainerComponent implements OnInit, OnDestroy {
     }
   }
 
-  onPickupOptionChange(option: PickupOption) {
-    this.intendedPickupLocationService.setPickupOption(
-      this.productCode,
-      option
-    );
-    if (option === 'delivery') {
-      return;
-    }
-    if (!this.displayNameIsSet) {
-      this.openDialog();
+  // TODO: Remove 'PickupOprion' argument type once 'a11yDialogTriggerRefocus' feature flag is removed.
+  onPickupOptionChange(
```

Comment:
This is another situation we should have the deprecation warning for anybody using the old `PickupOption` parameter.

Reply:
I've used overloading and left a JSDoc.